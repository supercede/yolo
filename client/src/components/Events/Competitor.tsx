import React from "react";
import { EventTeamProps } from "src/types";

export function EventCompetitor(props: EventTeamProps) {
  return (
    <li className="flex justify-between mt-1">
      <p className="text-xs text-white sm:text-sm" data-testid="competitor-name">{props.competitor?.name}</p>
      <p className="text-xs text-green sm:text-sm" data-testid="competitor-score">{props.competitor?.score}</p>
    </li>
  );
}
