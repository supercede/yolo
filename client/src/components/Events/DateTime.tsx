import { DateTimeProps } from "src/types";

export function EventDateTime(props: DateTimeProps) {
  const startDate = new Date(props.startTime).toLocaleDateString("en-GB");
  const startTime = new Date(props.startTime).toLocaleTimeString("en-GB", {
    hour12: false,
  });

  return (
    <p className="text-xs sm:text-sm text-grey">
      <span data-testid="event-start-date" data-test>
        {startDate.substring(0, startDate.length - 5)}{" "}
      </span>
      <span data-testid="event-start-time" data-test>
        {startTime.substring(0, startTime.length - 3)}
      </span>
    </p>
  );
}
