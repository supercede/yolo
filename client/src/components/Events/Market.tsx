import { EventContainerProps, EventMarketProps, Selection } from "src/types";

export function EventMarket(props: EventMarketProps) {
    return (
        <ul className="flex w-1/2 gap-2 p-2">
        {props?.markets?.[0] ? (
          props.markets[0]?.selections?.map((selection: Selection) => (
            <li
              key={selection.id}
              className="flex items-center justify-center w-full h-full text-white rounded bg-bg-blue"
              data-testid={`market-odds-${selection.id}`}
            >
              {selection.odds ? Number(selection.odds)?.toFixed(2) : "-"}
            </li>
          ))
        ) : (
          <div className="flex items-center justify-center w-full h-full text-white rounded bg-bg-blue">
            <p className="text-sm text-grey" data-testid="no-market">No market available</p>
          </div>
        )}
      </ul>
    )
}