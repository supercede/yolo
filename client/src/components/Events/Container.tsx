import { EventCompetitor } from "./Competitor";
import React from "react";
import { Competitor, EventContainerProps, Selection } from "src/types";
import { EventMarket } from "./Market";
import { EventDateTime } from "./DateTime";

export function EventContainer(props: EventContainerProps) {
  return (
    <li className="flex rounded-sm bg-blue">
      <div className="flex flex-col w-2/3 p-3">
        <ul className="flex flex-col pr-2 mt-1">
          {props.event.competitors.map((competitor: Competitor) => (
            <EventCompetitor competitor={competitor} key={competitor.id} />
          ))}
        </ul>
        <EventDateTime startTime={props.event.startTime} />
      </div>
      <EventMarket markets={props.event?.markets} />
    </li>
  );
}
