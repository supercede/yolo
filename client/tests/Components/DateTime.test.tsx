import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import { EventDateTime } from "@/components/Events/DateTime";
import data from "../data";

test("Test DateTime component", () => {
  const dateTime = data.event.payload.startTime;
  const { container } = render(
    <EventDateTime startTime={new Date(dateTime)} />
  );

  const startDate = new Date(dateTime).toLocaleDateString("en-GB");
  const startTime = new Date(dateTime).toLocaleTimeString("en-GB", {
    hour12: false,
  });
  expect(container.firstChild).toMatchSnapshot();
  expect(screen.getByTestId("event-start-date")).toHaveTextContent(
    new Date(dateTime)
      .toLocaleDateString("en-GB")
      .substring(0, startDate.length - 5)
  );
  expect(screen.getByTestId("event-start-time")).toHaveTextContent(
    new Date(dateTime)
      .toLocaleTimeString("en-GB", {
        hour12: false,
      })
      .substring(0, startTime.length - 3)
  );
});
