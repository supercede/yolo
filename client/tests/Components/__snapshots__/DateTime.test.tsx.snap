// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DateTime component 1`] = `
<p
  class="text-xs sm:text-sm text-grey"
>
  <span
    data-test="true"
    data-testid="event-start-date"
  >
    07/09
     
  </span>
  <span
    data-test="true"
    data-testid="event-start-time"
  >
    03:50
  </span>
</p>
`;
